### PR TITLE
DDPB-3356: Reinstates option to edit deputy email address for super a…

### PR DESCRIPTION
**Pull request by aiesoftware**
## Purpose
To allow super admins (only) to update the email of address of users, from the user edit page in admin.

Fixes DDPB-3356

## Approach
Added the `email` field to the `EditUserType` form class when the operating user has role `ROLE_SUPER_ADMIN`. I created a new `admin_edit_user` JMS group and validation group to separate it from the `admin_add_user` group.

## Checklist
- [ ] I have performed a self-review of my own code
- [ ] I have updated documentation (Confluence/ADR/tech debt doc) where relevant
- [ ] I have added tests to prove my work
- [ ] The product team have approved these changes

### Frontend
- [ ] I have run an in-browser accessibility test (e.g. WAVE, Lighthouse)
- [ ] There are no deprecated CSS classes noted in the profiler
- [ ] Translations are used and the profiler doesn't identify any missing

### CODE REVIEW

File: client/src/AppBundle/Resources/views/Admin/Index/editUser.html.twig [hunk 1]
```diff
@@ -22,6 +23,11 @@
 
     {{ form_input(form.firstname,'addUserForm.firstname') }}
     {{ form_input(form.lastname,'addUserForm.lastname') }}
+
+    {% if form.email is defined and app.user.roleName == class_const(UserClass, 'ROLE_SUPER_ADMIN') %}
```
Comment (gregtyler) [inline client/src/AppBundle/Resources/views/Admin/Index/editUser.html.twig:27]: Is the role check necessary? In other forms I've just used `if form.email is defined` and let the form handle permissions exclusively

File: behat/tests/features/UserManagement/editiingUserDetails.admin.feature [hunk 2]
```diff
@@ -0,0 +1,75 @@
+Feature: Editing Deputy details as an admin user
```
Comment (gregtyler) [inline behat/tests/features/UserManagement/editiingUserDetails.admin.feature:1]: The filename has an extra `i` in it, should be `editingUserDetails.admin.feature`

File: client/src/AppBundle/Resources/views/Admin/Index/editUser.html.twig (hunk 1 above)
Comment (aiesoftware) [reply client/src/AppBundle/Resources/views/Admin/Index/editUser.html.twig:27]: No not especially, happy to remove it.

File: behat/tests/features/UserManagement/editiingUserDetails.admin.feature (hunk 2 above)
Comment (aiesoftware) [reply behat/tests/features/UserManagement/editiingUserDetails.admin.feature:1]: 👀

**Pull request merged**